Add example on supporting both OpenTofu and Terraform versions to version-constraints.mdx (#2302)
opentofu/opentofu: website/docs/language/expressions/version-constraints.mdx

- Root modules should use a `~>` constraint to set both a lower and upper bound
  on versions for each provider they depend on.

### Supporting both OpenTofu and Terraform Versions

- When configuration needs to be supported by both OpenTofu and Terraform, for example in modules
  which need to be consumed by both engines, use a `providers.tofu` file to specify
  the OpenTofu version constraint, overriding the constraint defined in `providers.tf`.

>   [Read more about extension precedence](../../language/files/#extension-precedence).

Reviewer: I suggested to put a broken link in here, my bad :) Run website locally - the one below is correct:

```suggestion
  [Read more about extension precedence](../../files/#extension-precedence).
```

- Root modules should use a `~>` constraint to set both a lower and upper bound
  on versions for each provider they depend on.

### Supporting both OpenTofu and Terraform Versions

> - When code needs to support both OpenTofu and Terraform, for example in modules

Reviewer: ```suggestion
- When configuration needs to be supported by both OpenTofu and Terraform, for example in modules
```

  which need to be consumed by both engines, use a `providers.tofu` file to specify

>   the OpenTofu version constraint, overriding the constraint defined in `providers.tf`.

Reviewer: ```suggestion
  the OpenTofu version constraint, overriding the constraint defined in `providers.tf`.
  [Read more about extension precedence](../../language/files/#extension-precedence).
```
